### ISSUE TITLE
[BUG] TapNet TF: RP dimension uses series_length and CNN weights not shared (paper mismatch)

### ISSUE THREAD

**Issue by srupat** [sktime/sktime#9446]
**Describe the bug**
There are 3 critical issues in the TapNet estimator implementation in Tensorflow. 
- In the convolutional layers, the tf implementation uses kernel_size[0] everywhere, but according to the [original TapNet paper](https://xuczhang.github.io/papers/aaai20_tapnet_full.pdf), its appropriate to use all 3 of the specified kernel sizes.
- The paper mentions parameter sharing for the convolutional layers, however the tensorflow implementation does not implement this.
- When computing the random projection dimensions, it uses the series length, but later selects channels along axis 2. The correct way is to handle the case for `n_dims <= 1` and use just `n_dims` (channels) to calculate the rp dimensions. 

> This issue was identified while working on #9322 

**Additional context**
Reviewer comment: https://github.com/sktime/sktime/pull/9412#issuecomment-3937094030
TapNet paper link: https://xuczhang.github.io/papers/aaai20_tapnet_full.pdf
TapNet PyTorch implementation PR (with necessary changes): https://github.com/sktime/sktime/pull/9412

**Versions**
<details>

System:
    python: 3.12.7 | packaged by Anaconda, Inc. | (main, Oct  4 2024, 13:17:27) [MSC v.1929 64 bit (AMD64)]
executable: C:\ProgramData\anaconda3\python.exe
   machine: Windows-11-10.0.26200-SP0

Python dependencies:
          pip: 25.3
       sktime: 0.40.1
      sklearn: 1.7.0
       skbase: 0.13.1
        numpy: 1.26.4
        scipy: 1.16.0
       pandas: 2.3.3
   matplotlib: 3.9.2
       joblib: 1.5.1
huggingface-hub: 0.24.6
        numba: 0.62.1
     pmdarima: None
pytorch-forecasting: None
   skforecast: None
        skpro: None
statsforecast: None
  statsmodels: 0.14.2
 transformers: 4.44.2
      tsfresh: None
      tslearn: None
        torch: 2.9.1
   tensorflow: None
</details>

**Comment by AzamatMedetbekov** [comment]
Hi @srupat, can I work on this issue?

**Comment by srupat** [comment]
> Hi [@srupat](https://github.com/srupat), can I work on this issue?

Hello @AzamatMedetbekov . Sorry, I have raised a PR that will fix this issue since I am working on the PyTorch implementation of the same estimator.
